community/cypher/docs/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/DocumentingTestBase.scala

```diff
@@ -283,35 +279,58 @@ abstract class DocumentingTestBase extends JUnitSuite with DocumentationHelper w
     val testQuery = filePaths.foldLeft(query)( (acc, entry) => acc.replace(entry._1, entry._2))
     val docQuery = urls.foldLeft(query)( (acc, entry) => acc.replace(entry._1, entry._2))
 
-    try {
-      val result = fetchQueryResults(prepare, testQuery)
-      if (expectedException.isDefined) {
-        fail(s"Expected the test to throw an exception: $expectedException")
-      }
+    executeWithAllPlannersAndAssert(testQuery, assertions, expectedException,
+      dumpToFile(dir, writer, title, docQuery, optionalResultExplanation, text, _, consoleData), prepareForTest(title, prepare, dbPrepare))
+    match {
+      case Some(result) => dumpToFile(dir, writer, title, docQuery, optionalResultExplanation, text, result, consoleData)
+      case  None =>
+    }
 
-      dumpToFile(dir, writer, title, docQuery, optionalResultExplanation, text, result, consoleData)
-      if (graphvizExecutedAfter) {
-        dumpGraphViz(dir, graphvizOptions.trim)
-      }
+    if (graphvizExecutedAfter) {
+      dumpGraphViz(dir, graphvizOptions.trim)
+    }
+  }
 
-      db.inTx {
-        assertions.foreach(_.apply(result))
-      }
+  def prepareForTest(title: String, prepare: Option[() => Any], dbPrepare: GraphDatabaseAPI => Unit): Unit = {
+    prepare.foreach {
+      (prepareStep: () => Any) => prepareStep()
+    }
+    if (preparationQueries.nonEmpty) {
+      dumpPreparationQueries(preparationQueries, dir, title)
+      dumpPreparationGraphviz(dir, title, graphvizOptions)
+    }
+    dbPrepare(db)
+  }
 
-    } catch {
-      case e: CypherException if expectedException.nonEmpty =>
+  private def executeWithAllPlannersAndAssert(query: String, assertions: InternalExecutionResult => Unit,
+                                              expectedException: Option[ClassTag[_ <: CypherException]],
+                                              expectedCaught: CypherException => Unit,
+                                              prepareFunction: => Unit) = {
+    val params = Option(parameters).getOrElse(Map.empty)
+
+    // COST planner is default. Can't specify it without getting exception thrown if it's unavailable.
+    val planners = Seq("", "CYPHER PLANNER=rule ")
```

Comment:
Hm, didn't consider using both DP/IDP. Which one is default if nothing is specified by the query?